framework/source/class/qx/util/DynamicScriptLoader.js

```
@@ -0,0 +1,241 @@
+     EVENTS
+  *****************************************************************************
+  */
+
+  events: {
+    /**
+     * fired when a script is loaded successfully. The data contains 'script' and 'status' keys.
+     */
+    loaded: 'qx.event.type.Data',
+
+    /**
+     * fired when a specific script fails loading.  The data contains 'script' and 'status' keys.
+     */
+    failed: 'qx.event.type.Data',
+
+    /**
+     * fired when all given scripts are loaded, each time loadScriptsDynamic is called.
+     */
+    ready: 'qx.event.type.Event'
+  },
+
+  statics: {
+    /**
+     * Map of scripts being added at the present time. Key is script name; value is instance of this class which
+     * is loading it.
+     */
+    __IN_PROGRESS: {},
+     /**
+      * Map of scripts that have fully loaded. Key is script name; value is true
+      */
+    __LOADED: {}
+  },
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+
+  members: {
+
+    __QUEUE: null,
+    __LOADING: null,
+
+
+    /**
+     * Load scripts dynamically, typically used in the instance constructor
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     *
+     */
+    load: function(codeArr) {
+      codeArr.forEach(function(script) {
+         this.__QUEUE.push(script);
+      },this);
+      this.__loadScripts();
+    },
+
+
+    /**
+     * Chain loading scripts.
+     *
+     * Recursively called until the array of scripts is consumed
+     *
+     * @param codeArr {Array} an array with the uri names of the scripts
+     */
+    __loadScripts: function() {
+      var cl = qx.util.DynamicScriptLoader;
+      var script;
+      var dynLoader;
+      var id1, id2;
+      var uri;
+      var loader;
+
+      if (this.__LOADING === true) {
+        return;
```

start() is good, and I'd go farther than just returning false if start() has already been called. It's definitely a programmer error if it's called more than once, so I'd throw an error.